pkg/controller/cassandracluster/reconcile.go

```diff
@@ -180,8 +180,50 @@ func CheckNonAllowedRemoveDC(rcc *ReconcileCassandraCluster, cc *api.CassandraCl
 
 		//If we ask to remove only a rack, then it is not authorized
 		if dcsize == olddcsize {
-			logrus.WithFields(logrus.Fields{"cluster": cc.Name}).Warningf("The Operator has refused the Topology changed. You can only remove an entire DC, "+
-				"not only a Rack: %v restored to %v", cc.Spec.Topology, oldCRD.Spec.Topology)
+
+			//We need to check if the rack was existing or not
+
+			dcRackNameToDeleteList := cc.FixCassandraRackList(status)
+			if len(dcRackNameToDeleteList) > 0 {
+
+				if len(dcRackNameToDeleteList) > 1 {
+					logrus.WithFields(logrus.Fields{"cluster": cc.Name}).Warningf("The Operator has refused the Topology changed. You can only remove 1 rack if it is completely unshedulable, restored to %v", oldCRD.Spec.Topology)
+					cc.Spec.Topology = oldCRD.Spec.Topology
+					return true, api.ActionCorrectCRDConfig
+				}
+				for _, dcRackNameToDelete := range dcRackNameToDeleteList {
+
+						dcName, rackName := cc.GetDCAndRackFromDCRackName(dcRackNameToDelete)
+
+						listPod, err := rcc.ListPods(cc.Namespace, k8s.LabelsForCassandraDCRack(cc, dcName, rackName))
+						if err != nil {
+							logrus.WithFields(logrus.Fields{"cluster": cc.Name}).Warningf(
+								"The Operator has refused the Topology changed. We can't get associated pod in rack Rack: %v restored to %v", cc.Spec.Topology, oldCRD.Spec.Topology)
+							cc.Spec.Topology = oldCRD.Spec.Topology
+							return true, api.ActionCorrectCRDConfig
+						}
+						if len(listPod.Items) > 1{
+							logrus.WithFields(logrus.Fields{"cluster": cc.Name}).Warning("The Operator has refused the Topology changed. Too many pods in the Statefulset")
```

Comment:
@allamand, When could that one happen again ?

Reply:
oh it can happen if the scale down failed or did not happen ?

Reply:
it happens in case we added a new rack that has not been created yet because of lack of ressources. In this case it is allowed to remove the rack, but only if there is no pod at all really created in this rack